tooling/templatize/pkg/pipeline/arm.go

```diff
@@ -0,0 +1,135 @@
+package pipeline
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+)
+
+func (s *step) runArmStep(ctx context.Context, executionTarget *ExecutionTarget, options *PipelineRunOptions) error {
+	// Transform Bicep to ARM
+	deploymentProperties, err := transformBicepToARM(ctx, s.Parameters, options.Vars)
+	if err != nil {
+		return fmt.Errorf("failed to transform Bicep to ARM: %w", err)
+	}
+
+	// Create the deployment
+	deploymentName := s.Name
+	deployment := armresources.Deployment{
+		Properties: deploymentProperties,
+	}
+
+	// Ensure resourcegroup exists
+	err = s.ensureResourceGroupExists(ctx, executionTarget)
+	if err != nil {
+		return fmt.Errorf("failed to ensure resource group exists: %w", err)
+	}
+
+	// TODO handle dry-run
+
+	// Run deployment
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
An idea to make this testable, what do you off an interface for azure client, that you can mock in testing. See for instance https://github.com/app-sre/aws-resource-exporter/blob/master/pkg/awsclient/awsclient.go#L26 for an example on how we did in aws

Reply:
I would suggest end-to-end rather than an integration test. This is just glue code and if you mock out all of the interactions your integration test is low-value.

Reply:
i wrote down an action item